filesystem/unix_permission.go

```diff
@@ -7,43 +7,100 @@ package filesystem
 
 import (
 	"fmt"
+	"golang.org/x/sys/unix"
 	"os/user"
-	"syscall"
 )
 
+type FilePermission string
+
+const (
+	OwnerWrite  FilePermission = "OwnerWrite"
+	GroupWrite  FilePermission = "GroupWrite"
+	AnyoneWrite FilePermission = "AnyoneWrite"
+	OwnerRead   FilePermission = "OwnerRead"
+	AnyoneRead  FilePermission = "AnyoneRead"
+)
+
+var FilePermissionInHex = map[FilePermission]uint32{
+	OwnerWrite:  unix.S_IWUSR,
+	GroupWrite:  unix.S_IWGRP,
+	AnyoneWrite: unix.S_IWOTH,
+	OwnerRead:   unix.S_IRUSR,
+	AnyoneRead:  unix.S_IROTH,
+}
+
+func FileHasPermission(filePath string, permission FilePermission) (bool, error) {
+	fileStat, ok := GetFileStatPermission(filePath)
+	if ok != nil {
+		return false, ok
```

Comment:
ok is a confusing variable name. I would expect `ok` to be a bool and `true` to be the happy path.

I suggest:
```
fileStat, err := GetFileStatPermission(filePath)
if err != nil {
    return false, err
}
```

Reply:
good catch, i remember making the same change but i think i overrode it somehow